### PR TITLE
Simplify test 03246_range_literal_replacement_works

### PR DESCRIPTION
**Pull request by pamarcos**
Make the SQL compatible with older ClickHouse versions to backport it easily.

After merging https://github.com/ClickHouse/ClickHouse/pull/70122, I realized during the backports that the SQL could be simplified quite a bit because it failed to be parsed for 24.7 and 24.3

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: tests/queries/0_stateless/03246_range_literal_replacement_works.sql [hunk 1]
```diff
@@ -1,13 +1,5 @@
...
+CREATE TABLE 03246_range_literal_replacement_works (id UInt8) Engine=Memory;
 
-INSERT INTO my_table VALUES
-(
-CASE WHEN
-  (0 BETWEEN 0 AND 2) THEN 'ok' ELSE
-  'wat'
-END
-);
+INSERT INTO 03246_range_literal_replacement_works VALUES (1 BETWEEN 0 AND 2);
```
Comment (yariks5s) [inline tests/queries/0_stateless/03246_range_literal_replacement_works.sql:3]: Why do we need to change 0 -> 1 here?

**Comment by pamarcos** [comment]
@yariks5s pleas let me know if you're planning on making commits, because I have to sync whatever is here with 8 different branches for the backports, so every single commit makes it an extra effort since I expected to cherry-pick. Also, each test uses a ephimeral DB, so there's no need to add `IF NOT EXISTS` or drop the table at the end as I understand.

File: tests/queries/0_stateless/03246_range_literal_replacement_works.sql (hunk 1 above)
Comment (pamarcos) [reply tests/queries/0_stateless/03246_range_literal_replacement_works.sql:3]: A `1` to show success in the expected to me looks better in the expected file than a `0`. Anyway, the important part is that there's a range. This comes from https://github.com/ClickHouse/ClickHouse/pull/70122

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 22289f5c80ba8010d4bc13377f913c3832348f6b with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70176/22289f5c80ba8010d4bc13377f913c3832348f6b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/70160/2f4518fa66d4baa425fa6ef8479cb8f6debb369d/docs_check.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/9ccfb1235b840dbdf7c0d68b90a5c5bd24fedc75/unit_tests__release_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by pamarcos** [comment]
In debug, the previous `LOGICAL_ERROR`  [`e0d05bb` (#70122)](https://github.com/ClickHouse/ClickHouse/pull/70122/commits/e0d05bbadc92b9028f8b398f8fbdc87c15637d1e) would immediately abort the query. However, in release that was not the case because the exception was caught and a fallback logic applied.

Adding
```sql
SET input_format_values_interpret_expressions = 0;
SET input_format_values_accurate_types_of_literals = 0;
```

Ensures that even in release the template replacement fails before [`e0d05bb` (#70122)](https://github.com/ClickHouse/ClickHouse/pull/70122/commits/e0d05bbadc92b9028f8b398f8fbdc87c15637d1e) while now it works.

Please @yariks5s take a look whenever you can.

**Pull request merged**